From bb3f6d3a65b1589db6b98ff9cec0ca94f8efb438 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 27 May 2016 03:03:32 +0100
Subject: [PATCH 03/12] mutt-1.5.20-sendgroupreplyto

---
 init.h | 7 +++++++
 mutt.h | 1 +
 send.c | 8 ++++++++
 3 files changed, 16 insertions(+)

diff --git a/init.h b/init.h
index 2d06eef..659b55a 100644
--- a/init.h
+++ b/init.h
@@ -2885,6 +2885,13 @@ struct option_t MuttVars[] = {
   ** .pp
   ** Also see the $$force_name variable.
   */
+  { "send_group_reply_to",	DT_BOOL, R_NONE, OPTSENDGROUPREPLYTO, 0 },
+  /*
+  ** .pp
+  ** This variable controls how group replies are done.
+  ** When set, all recepients listet in "To:" are set in the
+  ** "To:" header again, else in the "CC", which is the default.
+  */
   { "score", 		DT_BOOL, R_NONE, OPTSCORE, 1 },
   /*
   ** .pp
diff --git a/mutt.h b/mutt.h
index 1d83793..3a49b19 100644
--- a/mutt.h
+++ b/mutt.h
@@ -483,6 +483,7 @@ enum
   OPTSAVEADDRESS,
   OPTSAVEEMPTY,
   OPTSAVENAME,
+  OPTSENDGROUPREPLYTO,
   OPTSCORE,
 #ifdef USE_SIDEBAR
   OPTSIDEBAR,
diff --git a/send.c b/send.c
index cea233d..7806b6b 100644
--- a/send.c
+++ b/send.c
@@ -624,8 +624,16 @@ int mutt_fetch_recips (ENVELOPE *out, ENVELOPE *in, int flags)
     if ((flags & SENDGROUPREPLY) && (!in->mail_followup_to || hmfupto != M_YES))
     {
       /* if(!mutt_addr_is_user(in->to)) */
+      if (option (OPTSENDGROUPREPLYTO))
+      {
+	rfc822_append (&out->to, in->to, 0);
+	rfc822_append (&out->cc, in->cc, 1);
+      }	  
+      else
+      {
       rfc822_append (&out->cc, in->to, 1);
       rfc822_append (&out->cc, in->cc, 1);
+      }
     }
   }
   return 0;
-- 
2.7.4

