From f84db73d80000ab78a024de66cfe0428b3930565 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 27 May 2016 03:03:48 +0100
Subject: [PATCH 04/13] mutt-1.5.15-wrapcolumn

---
 init.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/init.h b/init.h
index ae28ca8..b54d000 100644
--- a/init.h
+++ b/init.h
@@ -4021,7 +4021,8 @@ struct option_t MuttVars[] = {
   ** When \fIset\fP, mutt will weed headers when displaying, forwarding,
   ** printing, or replying to messages.
   */
-  { "wrap",             DT_NUM,  R_PAGER, UL &Wrap, 0 },
+  { "wrap",		DT_NUM,  R_PAGER, UL &Wrap, 0 },
+  { "wrapcolumn",	DT_SYN,  R_NONE, UL "wrap", 0 },
   /*
   ** .pp
   ** When set to a positive value, mutt will wrap text at $$wrap characters.
-- 
2.8.2

