From 67ccf7ef98cfdb585e055529c40c9756815f577d Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 27 May 2016 03:08:08 +0100
Subject: [PATCH 08/12] patch-1.5.24.vk.pgp_verbose_mime

---
 PATCHES   |  3 ++-
 globals.h |  2 ++
 init.h    | 12 ++++++++++++
 pgp.c     |  3 ++-
 4 files changed, 18 insertions(+), 2 deletions(-)

diff --git a/PATCHES b/PATCHES
index 2b899a9..daf8551 100644
--- a/PATCHES
+++ b/PATCHES
@@ -20,4 +20,5 @@ patch-status-color-neo-git
 patch-timeout-neo-git
 patch-tls-sni-neo-git
 patch-trash-neo-git
-+patch-1.5.9.aw.listreply.1
+patch-1.5.9.aw.listreply.1
+patch-1.5.3.vk.pgp_verbose_mime
diff --git a/globals.h b/globals.h
index d4e7fa6..0d97519 100644
--- a/globals.h
+++ b/globals.h
@@ -271,6 +271,8 @@ WHERE REGEXP PgpDecryptionOkay;
 WHERE char *PgpSignAs;
 WHERE short PgpTimeout;
 WHERE char *PgpEntryFormat;
+WHERE char *PgpMimeSignatureFilename;
+WHERE char *PgpMimeSignatureDescription;
 WHERE char *PgpClearSignCommand;
 WHERE char *PgpDecodeCommand;
 WHERE char *PgpVerifyCommand;
diff --git a/init.h b/init.h
index 95412b3..a6a4c16 100644
--- a/init.h
+++ b/init.h
@@ -3205,6 +3205,18 @@ struct option_t MuttVars[] = {
   ** a line quoted text if it also matches $$smileys. This mostly
   ** happens at the beginning of a line.
   */
+  { "pgp_mime_signature_filename", DT_STR, R_NONE, UL &PgpMimeSignatureFilename, UL "signature.asc"},
+  /*
+  ** .pp
+  ** This option sets the filename used for signature parts in PGP/MIME
+  ** signed messages.
+  */
+  { "pgp_mime_signature_description", DT_STR, R_NONE, UL &PgpMimeSignatureDescription, UL "Digital signature"},
+  /*
+  ** .pp
+  ** This option sets the Content-Description used for signature parts in
+  ** PGP/MIME signed messages.
+  */
 
 
 
diff --git a/pgp.c b/pgp.c
index fb7be4c..0caff87 100644
--- a/pgp.c
+++ b/pgp.c
@@ -1261,7 +1261,8 @@ BODY *pgp_sign_message (BODY *a)
   t->disposition = DISPNONE;
   t->encoding = ENC7BIT;
   t->unlink = 1; /* ok to remove this file after sending. */
-  mutt_set_parameter ("name", "signature.asc", &t->parameter);
+  mutt_set_parameter ("name", PgpMimeSignatureFilename, &t->parameter);
+  t->description = safe_strdup (PgpMimeSignatureDescription);
 
   return (a);
 }
-- 
2.7.4

