From 85b2731d26be79a3efa00b9cdaf59bb2527a27d0 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 27 May 2016 03:08:25 +0100
Subject: [PATCH 09/12] widechar.sidebar

---
 handler.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/handler.c b/handler.c
index 6068291..274fedc 100644
--- a/handler.c
+++ b/handler.c
@@ -780,7 +780,7 @@ static void enriched_set_flags (const wchar_t *tag, struct enriched_state *stte)
 static int text_enriched_handler (BODY *a, STATE *s)
 {
   enum {
-    TEXT, LANGLE, TAG, BOGUS_TAG, NEWLINE, ST_EOF, DONE
+    TEXT, TEXT_GOTWC, LANGLE, TAG, BOGUS_TAG, NEWLINE, ST_EOF, DONE
   } state = TEXT;
 
   long bytes = a->length;
@@ -809,7 +809,7 @@ static int text_enriched_handler (BODY *a, STATE *s)
   {
     if (state != ST_EOF)
     {
-      if (!bytes || (wc = fgetwc (s->fpin)) == WEOF)
+      if (state != TEXT_GOTWC && (!bytes || (fscanf (s->fpin, "%lc", &wc)) == EOF))
 	state = ST_EOF;
       else
 	bytes--;
@@ -817,6 +817,8 @@ static int text_enriched_handler (BODY *a, STATE *s)
 
     switch (state)
     {
+      case TEXT_GOTWC:
+       state = TEXT;
       case TEXT :
 	switch (wc)
 	{
@@ -877,9 +879,8 @@ static int text_enriched_handler (BODY *a, STATE *s)
 	  enriched_flush (&stte, 1);
 	else
 	{
-	  ungetwc (wc, s->fpin);
 	  bytes++;
-	  state = TEXT;
+	  state = TEXT_GOTWC;
 	}
 	break;
 
@@ -1567,7 +1568,7 @@ void mutt_decode_attachment (BODY *b, STATE *s)
  * strip all trailing spaces to improve interoperability;
  * if $text_flowed is unset, simply verbatim copy input
  */
-static int text_plain_handler (BODY *b, STATE *s)
+static int text_plain_handler (BODY *b __attribute__((unused)), STATE *s)
 {
   char *buf = NULL;
   size_t l = 0, sz = 0;
-- 
2.9.0

