From 9730ca35449837565be41333f31bc78bc35bfd07 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Wed, 1 Jun 2016 21:39:26 +0100
Subject: [PATCH 5/5] mutt-1.6.0-syncdebug

---
 imap/imap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/imap/imap.c b/imap/imap.c
index feb06ef..4ed560a 100644
--- a/imap/imap.c
+++ b/imap/imap.c
@@ -1133,7 +1133,7 @@ static int sync_helper (IMAP_DATA* idata, int right, int flag, const char* name)
   if (!idata->ctx)
     return -1;
 
-  if (!mutt_bit_isset (idata->ctx->rights, right))
+  if (!idata->ctx || !mutt_bit_isset (idata->ctx->rights, right))
     return 0;
 
   if (right == M_ACL_WRITE && !imap_has_flag (idata->flags, name))
-- 
2.7.4

