From 1ca4aeabfc36950b72b8acde783145520f455030 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 6 May 2016 00:23:57 +0100
Subject: [PATCH 03/19] features/compressed-folders.debian

---
 config-debian.h | 4 ++++
 1 file changed, 4 insertions(+)
 create mode 100644 config-debian.h

diff --git a/config-debian.h b/config-debian.h
new file mode 100644
index 0000000..fdf5322
--- /dev/null
+++ b/config-debian.h
@@ -0,0 +1,4 @@
+#ifndef USE_COMPRESSED
+#error "You forgot to update debian/patches/autotools-update."
+#error "Run sh debian/update-autotools.sh"
+#endif
-- 
2.7.4

