From 43c380811ed3f8b1d163fec5d226a1bf2505e8ab Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 6 May 2016 00:23:59 +0100
Subject: [PATCH 08/22] debian-specific/dont_document_not_present_features

---
 functions.h | 2 +-
 init.h      | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/functions.h b/functions.h
index 5747425..250cab0 100644
--- a/functions.h
+++ b/functions.h
@@ -35,7 +35,7 @@
 
 #ifdef _MAKEDOC
 # include "config.h"
-# include "doc/makedoc-defs.h"
+/* # include "doc/makedoc-defs.h" include only documented options for Debian */
 #endif
 
 const struct binding_t OpGeneric[] = { /* map: generic */
diff --git a/init.h b/init.h
index 8a21b9c..d87e31d 100644
--- a/init.h
+++ b/init.h
@@ -19,7 +19,7 @@
 
 #ifdef _MAKEDOC
 # include "config.h"
-# include "doc/makedoc-defs.h"
+/* # include "makedoc-defs.h" include only documented options for Debian */
 #else
 # include "sort.h"
 #endif
-- 
2.8.2

