From 865aa746b78f25ac11b91b353b6ab556ef8df4c9 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 6 May 2016 00:23:59 +0100
Subject: [PATCH 08/19] debian-specific/dont_document_not_present_features

---
 functions.h | 2 +-
 init.h      | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/functions.h b/functions.h
index dad63b6..52b0179 100644
--- a/functions.h
+++ b/functions.h
@@ -32,7 +32,7 @@
 
 #ifdef _MAKEDOC
 # include "config.h"
-# include "doc/makedoc-defs.h"
+/* # include "doc/makedoc-defs.h" include only documented options for Debian */
 #endif
 
 const struct binding_t OpGeneric[] = { /* map: generic */
diff --git a/init.h b/init.h
index eba1f83..2fbcc3a 100644
--- a/init.h
+++ b/init.h
@@ -19,7 +19,7 @@
 
 #ifdef _MAKEDOC
 # include "config.h"
-# include "doc/makedoc-defs.h"
+/* # include "makedoc-defs.h" include only documented options for Debian */
 #else
 # include "sort.h"
 #endif
-- 
2.7.4

