From 71d37e3b798c2b56fb5bf24b7fdc0653a9aed39e Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 6 May 2016 00:24:01 +0100
Subject: [PATCH 13/19] misc/define-pgp_getkeys_command

---
 contrib/gpg.rc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/contrib/gpg.rc b/contrib/gpg.rc
index 2ef0ae2..53f3965 100644
--- a/contrib/gpg.rc
+++ b/contrib/gpg.rc
@@ -72,6 +72,8 @@ set pgp_list_secring_command="gpg --no-verbose --batch --quiet --with-colons --w
 
 # fetch keys
 # set pgp_getkeys_command="pkspxycwrap %r"
+# This will work when #172960 will be fixed upstream
+# set pgp_getkeys_command="gpg --recv-keys %r"
 
 # pattern for good signature - may need to be adapted to locale!
 
-- 
2.7.4

