From 01f042bcbd7ad7fce151a997ef19d4de382ac87c Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 6 May 2016 00:24:11 +0100
Subject: [PATCH 21/22] __separator__mutt.org

---
 .mutt.org | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 .mutt.org

diff --git a/.mutt.org b/.mutt.org
new file mode 100644
index 0000000..d0d32f3
--- /dev/null
+++ b/.mutt.org
@@ -0,0 +1 @@
+# dummy patch to serve as marker separating patches for mutt-patched
-- 
2.7.4

