From 879fe0c3cfee4b86506a74c0184bbb7b252585a8 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 27 May 2016 15:16:58 +0100
Subject: [PATCH 03/21] vvv.quote

---
 PATCHES   |  1 +
 handler.c | 28 ++++++++++++++++++++++++++--
 init.h    | 13 +++++++++++++
 mutt.h    |  2 ++
 4 files changed, 42 insertions(+), 2 deletions(-)

diff --git a/PATCHES b/PATCHES
index fd6fa32..441a58b 100644
--- a/PATCHES
+++ b/PATCHES
@@ -1,3 +1,4 @@
+vvv.quote
 patch-ifdef-neo-UNKNOWN
 patch-initials-neo-UNKNOWN
 patch-trash-neo-UNKNOWN
diff --git a/handler.c b/handler.c
index 4944d49..dfeb96e 100644
--- a/handler.c
+++ b/handler.c
@@ -1581,7 +1581,31 @@ static int text_plain_handler (BODY *b, STATE *s)
 	buf[--l] = 0;
     }
     if (s->prefix)
-      state_puts (s->prefix, s);
+    {
+      int i;
+      char qbuf[2 * LONG_STRING];
+      int j = 0, offset = 0;
+      regmatch_t pmatch[1];
+
+      while (regexec ((regex_t *) QuoteRegexp.rx, &buf[offset], 1, pmatch, 0) == 0)
+	offset += pmatch->rm_eo;
+
+      if (!option (OPTQUOTEEMPTY) && !*buf)
+	strcpy (qbuf, buf);
+      else if (option (OPTQUOTEQUOTED) && offset)
+      {
+	for (i = 0; i < offset; i++)
+	  if (buf[i] != ' ')
+	    j = i;
+	strncpy (qbuf, buf, j + 1);
+	strcpy (qbuf + j + 1, buf + j);
+      }
+      else
+	snprintf (qbuf, sizeof (qbuf), "%s%s", NONULL(s->prefix), buf);
+
+      state_puts (qbuf, s);
+    }
+    else
     state_puts (buf, s);
     state_putc ('\n', s);
   }
@@ -1737,7 +1761,7 @@ int mutt_body_handler (BODY *b, STATE *s)
        */
       if ((WithCrypto & APPLICATION_PGP) && mutt_is_application_pgp (b))
 	handler = crypt_pgp_application_pgp_handler;
-      else if (option(OPTREFLOWTEXT) && ascii_strcasecmp ("flowed", mutt_get_parameter ("format", b->parameter)) == 0)
+      else if (option(OPTREFLOWTEXT) && ascii_strcasecmp ("flowed", mutt_get_parameter ("format", b->parameter)) == 0 && !s->prefix)
 	handler = rfc3676_handler;
       else
 	handler = text_plain_handler;
diff --git a/init.h b/init.h
index 71d170c..422c430 100644
--- a/init.h
+++ b/init.h
@@ -2332,6 +2332,19 @@ struct option_t MuttVars[] = {
   ** have no effect, and if it is set to \fIask-yes\fP or \fIask-no\fP, you are
   ** prompted for confirmation when you try to quit.
   */
+  { "quote_empty",	DT_BOOL, R_NONE, OPTQUOTEEMPTY, 1 },
+  /*
+  ** .pp
+  ** Controls whether or not empty lines will be quoted using
+  ** ``$indent_string''.
+  */
+  { "quote_quoted",	DT_BOOL, R_NONE, OPTQUOTEQUOTED, 0 },
+  /*
+  ** .pp
+  ** Controls how quoted lines will be quoted. If set, one quote
+  ** character will be added to the end of existing prefix.  Otherwise,
+  ** quoted lines will be prepended by ``$indent_string''.
+  */
   { "quote_regexp",	DT_RX,	 R_PAGER, UL &QuoteRegexp, UL "^([ \t]*[|>:}#])+" },
   /*
   ** .pp
diff --git a/mutt.h b/mutt.h
index 8d376b5..164069a 100644
--- a/mutt.h
+++ b/mutt.h
@@ -420,6 +420,8 @@ enum
   OPTPRINTDECODE,
   OPTPRINTSPLIT,
   OPTPROMPTAFTER,
+  OPTQUOTEEMPTY,
+  OPTQUOTEQUOTED,
   OPTREADONLY,
   OPTREFLOWSPACEQUOTES,
   OPTREFLOWTEXT,
-- 
2.8.2

