From 7b470a163d19ec0e5b856265329fe90fb1d3b2f9 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 27 May 2016 15:18:21 +0100
Subject: [PATCH 06/21] smime-outlook

---
 crypt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crypt.c b/crypt.c
index 570cc36..83bbe96 100644
--- a/crypt.c
+++ b/crypt.c
@@ -493,7 +493,7 @@ int mutt_is_application_smime (BODY *m)
     {
       len++;
       if (!ascii_strcasecmp ((t+len), "p7m"))
-#if 0
+#if 1
        return SMIMEENCRYPT;
 #else
       /* Not sure if this is the correct thing to do, but 
-- 
2.8.2

