From cd94df70e401326269e1a9af24b61331f16b4a16 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 27 May 2016 15:18:48 +0100
Subject: [PATCH 07/21] browser

---
 browser.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/browser.c b/browser.c
index 46919a0..62fae18 100644
--- a/browser.c
+++ b/browser.c
@@ -81,6 +81,12 @@ static int browser_compare_subject (const void *a, const void *b)
   struct folder_file *pa = (struct folder_file *) a;
   struct folder_file *pb = (struct folder_file *) b;
 
+  /* Always keep '..' in first menu position */
+  if (!strncmp(pa->name, "..", 2))
+	  return(-1);
+  if (!strncmp(pb->name, "..", 2))
+	  return(1);
+
   int r = mutt_strcoll (pa->name, pb->name);
 
   return ((BrowserSort & SORT_REVERSE) ? -r : r);
@@ -91,6 +97,12 @@ static int browser_compare_date (const void *a, const void *b)
   struct folder_file *pa = (struct folder_file *) a;
   struct folder_file *pb = (struct folder_file *) b;
 
+  /* Always keep '..' in first menu position */
+  if (!strncmp(pa->name, "..", 2))
+	  return(-1);
+  if (!strncmp(pb->name, "..", 2))
+	  return(1);
+
   int r = pa->mtime - pb->mtime;
 
   return ((BrowserSort & SORT_REVERSE) ? -r : r);
@@ -101,6 +113,12 @@ static int browser_compare_size (const void *a, const void *b)
   struct folder_file *pa = (struct folder_file *) a;
   struct folder_file *pb = (struct folder_file *) b;
 
+  /* Always keep '..' in first menu position */
+  if (!strncmp(pa->name, "..", 2))
+	  return(-1);
+  if (!strncmp(pb->name, "..", 2))
+	  return(1);
+
   int r = pa->size - pb->size;
 
   return ((BrowserSort & SORT_REVERSE) ? -r : r);
-- 
2.8.2

