From 3db1188ce711b29b70a8e8023b9c3c94cf6d4b1b Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 27 May 2016 15:20:06 +0100
Subject: [PATCH 10/21] doc-manual.xml.head

---
 doc/manual.xml.head | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/manual.xml.head b/doc/manual.xml.head
index 7f6988e..6e46124 100644
--- a/doc/manual.xml.head
+++ b/doc/manual.xml.head
@@ -1,6 +1,6 @@
 <?xml version="1.0" standalone="no"?>
 <!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
-  "http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd">
+  "/usr/local/share/xml/docbook/4.2/docbookx.dtd">
 <book>
 
 <bookinfo>
-- 
2.8.2

