From 1d2cb96534be87904d996d625ce74109f6792042 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 27 May 2016 15:21:22 +0100
Subject: [PATCH 11/21] examples

---
 INSTALL             |  4 ++--
 contrib/Makefile.am |  7 ++++---
 init.h              | 10 ++++------
 3 files changed, 10 insertions(+), 11 deletions(-)

diff --git a/INSTALL b/INSTALL
index 2242dd2..8de8360 100644
--- a/INSTALL
+++ b/INSTALL
@@ -228,13 +228,13 @@ If you decide to use your system's iconv implementation, you may
 need to tell mutt about implementation-defined names for some
 character sets.  Sample configuration files for various systems can
 be found in the directory contrib/iconv/ in this source
-distribution, and will be installed in the samples/iconv directory
+distribution, and will be installed in the share/examples/mutt/iconv directory
 as part of mutt's documentation.
 
 In order to use these sample configuration files, just put a line
 like
 
-     source /usr/local/doc/mutt/samples/iconv/iconv.osf1-4.0d.rc
+     source /usr/local/share/examples/mutt/iconv/iconv.osf1-4.0d.rc
 
 into your system's global Muttrc, which normally resides in /etc or
 /usr/local/etc.
diff --git a/contrib/Makefile.am b/contrib/Makefile.am
index 34b09e3..5cfc462 100644
--- a/contrib/Makefile.am
+++ b/contrib/Makefile.am
@@ -1,6 +1,7 @@
 # $Id$
 
 subdir = contrib
+samplesdir = @samplesdir@
 
 SAMPLES = Mush.rc Pine.rc gpg.rc pgp2.rc pgp6.rc Tin.rc \
 	sample.muttrc  sample.mailcap sample.muttrc-tlr \
@@ -14,12 +15,12 @@ EXTRA_DIST = language.txt language50.txt  \
 	iconv/make.sh
 
 install-data-local:
-	$(MKDIR_P) $(DESTDIR)$(docdir)/samples $(DESTDIR)$(docdir)/samples/iconv
+	$(MKDIR_P) $(DESTDIR)$(samplesdir) $(DESTDIR)$(samplesdir)/iconv
 	for f in $(SAMPLES) ; do \
-		$(INSTALL) -m 644 $(srcdir)/$$f $(DESTDIR)$(docdir)/samples ;	\
+		$(INSTALL) -m 644 $(srcdir)/$$f $(DESTDIR)$(samplesdir) ;	\
 	done
 	for f in $(srcdir)/iconv/*.rc ; do					\
-		$(INSTALL) -m 644 $$f $(DESTDIR)$(docdir)/samples/iconv	  ;	\
+		$(INSTALL) -m 644 $$f $(DESTDIR)$(samplesdir)/iconv  ;		\
 	done
 
 uninstall-local:
diff --git a/init.h b/init.h
index 510a3d9..b621200 100644
--- a/init.h
+++ b/init.h
@@ -1810,9 +1810,8 @@ struct option_t MuttVars[] = {
   ** .pp
   ** For examples on how to configure these formats for the various versions
   ** of PGP which are floating around, see the pgp and gpg sample configuration files in
-  ** the \fCsamples/\fP subdirectory which has been installed on your system
-  ** alongside the documentation.
-  ** (PGP only)
+  ** the \fCshare/examples/mutt/\fP subdirectory which has been installed on
+  **  your system. (PGP only)
   */
   { "pgp_decrypt_command", 	DT_STR, R_NONE, UL &PgpDecryptCommand, 0},
   /*
@@ -2958,9 +2957,8 @@ struct option_t MuttVars[] = {
   ** .de
   ** .pp
   ** For examples on how to configure these formats, see the \fCsmime.rc\fP in
-  ** the \fCsamples/\fP subdirectory which has been installed on your system
-  ** alongside the documentation.
-  ** (S/MIME only)
+  ** the \fCshare/examples/mutt/\fP subdirectory which has been installed on
+  ** your system. (S/MIME only)
   */
   { "smime_decrypt_use_default_key",	DT_BOOL, R_NONE, OPTSDEFAULTDECRYPTKEY, 1 },
   /*
-- 
2.7.4

