From adbe1c25f2e251482707148ac8eb41a4f0a2d9bd Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 27 May 2016 15:21:48 +0100
Subject: [PATCH 13/21] m4-gssapi.m4

---
 m4/gssapi.m4 | 1 +
 1 file changed, 1 insertion(+)

diff --git a/m4/gssapi.m4 b/m4/gssapi.m4
index 0e0194e..80f14ed 100644
--- a/m4/gssapi.m4
+++ b/m4/gssapi.m4
@@ -27,6 +27,7 @@ AC_DEFUN([MUTT_AM_PATH_GSSAPI],
     GSSAPI_LIBS="`$KRB5CFGPATH --libs gssapi`"
     case "`$KRB5CFGPATH --version`" in
       "Kerberos 5 "*)	GSSAPI_IMPL="MIT";;
+      "FreeBSD heimdal"*)		GSSAPI_IMPL="Heimdal";;
       ?eimdal*)		GSSAPI_IMPL="Heimdal";;
       *)		GSSAPI_IMPL="Unknown";;
    esac
-- 
2.7.4

