From 54651936376f156ee0aaf67d451f76ba167e72bd Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Sun, 29 May 2016 13:34:29 +0100
Subject: [PATCH 3/5] mutt-1.6.0-syncdebug

---
 imap/imap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/imap/imap.c b/imap/imap.c
index 1b63b3a..ea3fcb1 100644
--- a/imap/imap.c
+++ b/imap/imap.c
@@ -1125,7 +1125,7 @@ static int sync_helper (IMAP_DATA* idata, int right, int flag, const char* name)
   if (!idata->ctx)
     return -1;
 
-  if (!mutt_bit_isset (idata->ctx->rights, right))
+  if (!idata->ctx || !mutt_bit_isset (idata->ctx->rights, right))
     return 0;
 
   if (right == M_ACL_WRITE && !imap_has_flag (idata->flags, name))
-- 
2.7.4

